### ISSUE TITLE
llvm: patch doesn't work with HEAD

### ISSUE THREAD

**Issue by Char-Aznable** [Homebrew/linuxbrew-core#12859]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
brew install llvm --HEAD
- What happened (include command output)
Failed with problem patching
- What you expected to happen
installation succeeds
- Step-by-step reproduction instructions (by running `brew install` commands)
brew install llvm --HEAD

Basically, the 
```
       url "https://gist.githubusercontent.com/iMichka/027fd3d17b4c729e73a190ae29e44b47/raw/a88c628f28ca9cd444cc3771072260fe46ff8a29/llvm7.patch?full_index=
          sha256 "8db2acff4fbe0533667c9a0527a6a180fd2a84daea4271665fd42f88a08eaa86"                                                                            
        end unless OS.mac? 
```
doesn't apply to the latest llvm dev any more. I removed these and llvm head compiles fine.

**Comment by iMichka** [comment]
Yes. This may not apply anymore for head. 

Head is a moving target. We may fix the patch, and it may break again in two days.

I would be willing to accept a patch for head as a pull request, but have no time to invest on maintaining patches for intermediate llvm versions.

**Comment by Char-Aznable** [comment]
I meant the patch for head is no longer necessary. We should just remove it.

**Comment by iMichka** [comment]
Maybe @xu-cheng has some time, he initially wrote the patch and I have just been adapting it for each llvm release since then.

**Comment by xu-cheng** [comment]
Could you try this new patch: https://github.com/xu-cheng/clang/commit/fd8e019e1ca9ec63b259494e4ccacc20bd896fbe
